checkstyle-tester/diff.groovy

```diff
@@ -358,16 +358,34 @@ def getCloneCmd(repoType, repoUrl, srcDestinationDir) {
     return cloneCmd
 }
 
+def getCloneShallowCmd(repoType, repoUrl, srcDestinationDir, branchOrTag) {
+    def cloneCmd = ''
+    switch (repoType) {
+        case 'git':
+            cloneCmd = "git clone --depth 1 --branch $branchOrTag $repoUrl $srcDestinationDir"
+            break
+        default:
+            throw new IllegalArgumentException("Error! Unknown $repoType repository.")
+    }
+    return cloneCmd
+}
+
 def cloneRepository(repoName, repoType, repoUrl, commitId, srcDir) {
     def srcDestinationDir = getOsSpecificPath("$srcDir", "$repoName")
     if (!Files.exists(Paths.get(srcDestinationDir))) {
-        def cloneCmd = getCloneCmd(repoType, repoUrl, srcDestinationDir)
+        def cloneCmd
+        if (isGitSha(commitId)) {
+            cloneCmd = getCloneCmd(repoType, repoUrl, srcDestinationDir)
+        } else {
+            cloneCmd = getCloneShallowCmd(repoType, repoUrl, srcDestinationDir, commitId)
+        }
         println "Cloning $repoType repository '$repoName' to $srcDestinationDir folder ..."
+        println "Cloning command: $cloneCmd"
         executeCmdWithRetry(cloneCmd)
         println "Cloning $repoType repository '$repoName' - completed\n"
     }
 
-    if (commitId && commitId != '') {
+    if (commitId && commitId != '' && isGitSha(commitId)) {
```

Comment:
```suggestion
    if (isGitSha(commitId)) {
```

Reply:
Wait, this change (current, and proposed) breaks `hg` support: https://github.com/checkstyle/contribution/blob/0abf4d92e996e959863477af18c3066ad3e67a95/checkstyle-tester/diff.groovy#L713

Reply:
We don't support hg, let's forget about it. I can remove it in next PR. We did not use it for many years.

Reply:
https://github.com/checkstyle/contribution/issues/822
https://github.com/checkstyle/contribution/pull/823

Reply:
Initial suggestion at https://github.com/checkstyle/contribution/pull/819/files#r1424827390 still stands.

Reply:
applied, tested.

```
diff --git a/checkstyle-tester/projects-to-test-on.properties b/checkstyle-tester/projects-to-test-on.properties
index bed1425..9d259dd 100644
--- a/checkstyle-tester/projects-to-test-on.properties
+++ b/checkstyle-tester/projects-to-test-on.properties
@@ -27,7 +27,7 @@ guava|git|https://github.com/google/guava|v28.2||
 
 # Those projects are quite old and have lot of legacy code
 #apache-ant|git|https://github.com/apache/ant|ANT_194|**/apache-ant/src/tests/**/*,**/apache-ant/src/etc/testcases/
-#apache-jsecurity|git|https://github.com/apache/jsecurity|c2ac5b90a467aedb04b52ae50a99e83207d847b3||
+apache-jsecurity|git|https://github.com/apache/jsecurity|c2ac5b90a467aedb04b52ae50a99e83207d847b3||
 #android-launcher|git|https://github.com/android/platform_packages_apps_launcher|android-2.1_r2.1p2||
 #apache-struts|git|https://github.com/apache/struts.git|master|**/apache-struts/**/resources/**/*
```

cmd:
```
groovy diff.groovy --localGitRepo ~/java/github/romani/checkstyle \
  --baseBranch master --patchBranch plan3d/Issue_3132 --config my_check.xml \
  --listOfProjects projects-to-test-on.properties
```
[testing.txt](https://github.com/checkstyle/contribution/files/13674019/testing.txt)